File: Client/src/types/export-options.ts




export interface ExportOptions {
  format: string; // e.g. 'csv', 'json', 'xml'
  scope: string;
  startDate?: string; // ISO string
  endDate?: string; // ISO string
  includePerformanceMetrics: boolean;
  includeUmbracoMetrics: boolean;
  includeActiveRequests: boolean;
  timezone: string;
}

export interface ExportResult {
  fileName: string;
  contentType: string;
  data: ArrayBuffer;
  sizeBytes: number;
  generatedAt: string;
}

export interface ExportProgress {
  status: 'idle' | 'preparing' | 'exporting' | 'downloading' | 'complete' | 'error';
  progress: number; // 0-100
  message?: string;
  error?: string;
}

File: Models/ExportOptions.cs
namespace UmbMetrics.Models;

public class ExportOptions
{
    public ExportFormat Format { get; set; } = ExportFormat.Json;
    public ExportScope Scope { get; set; } = ExportScope.Current;
    public DateTime? StartDate { get; set; }
    public DateTime? EndDate { get; set; }
    public bool IncludePerformanceMetrics { get; set; } = true;
    public bool IncludeUmbracoMetrics { get; set; } = true;
    public bool IncludeActiveRequests { get; set; } = false;
    public string Timezone { get; set; } = "UTC";
}